include/zephyr/mgmt/mcumgr/mgmt/mgmt.h

```diff
@@ -120,6 +120,13 @@ void mgmt_register_group(struct mgmt_group *group);
  */
 void mgmt_unregister_group(struct mgmt_group *group);
 
+/**
+ * @brief Get list of registered groups
+ *
+ * @return Pointer to list of registered groups
+ */
+sys_slist_t *mgmt_get_group_list();
```

Comment:
This should be a function that calls a callback multiple times, once per group, and returns the group object, it should not be just returning a list directly or using sys_slist_t when the type should be the mgmt_group type (should also be const in the callback)